rtbkit/common/exchange_connector.h

```diff
@@ -231,6 +231,29 @@ struct ExchangeConnector: public ServiceBase {
     getCreativeCompatibility(const Creative & creative,
                              bool includeReasons) const;
 
+    /** Structure is used to determine if the bid is valid.
+    */
+    struct BidValidity {
+        BidValidity()
+            :isValidbid(true)
+        {
+        }
+
+        void setInvalidBid(const std::string &reason) {
```

Comment:
One more small detail. I think I would provide a `setValidBid()` counterpart function that will only set the boolean flag to `true`. When you default-construct a `BidValidity` object, you do not really know it by default it is considered valid or not, so having a `setValidBid()` function is more expressive:

```cpp
BidValidity result;
result.setValidBid();
```